datafusion/functions-array/src/range.rs

```diff
@@ -57,6 +56,7 @@ impl Range {
                     TypeSignature::Exact(vec![Int64, Int64]),
                     TypeSignature::Exact(vec![Int64, Int64, Int64]),
                     TypeSignature::Exact(vec![Date32, Date32, Interval(MonthDayNano)]),
+                    TypeSignature::Any(3),
```

Comment:
Does this mean we need to find a new kind of signature that accepts either one exact type or  null 🤔

Reply:
I think it would be better to have a new Signature. Maybe I can add more signatures in the following PR along with dealing #10200

Reply:
I'm thinking of something like
```
        TypeSignature::ExactOrNull(valid_types) => {
            current_types.iter().map(|t| {
                if t != &DataType::Null {
                    valid_types.clone()
                } else {
                    vec![DataType::Null]
                }
            })
        }
```

Reply:
Yeah, that would be fine. I'll add a ticket to record it